### ISSUE TITLE
🚨 Route with parameter matches request URL without parameter

### ISSUE THREAD

**Issue by dotvirus** [eon-web/eon#4]
**Describe the bug**
See title

**To Reproduce**
```javascript
const eon = require("eonjs");

const app = eon(8080);

app.get("/:name").json((req, res) => {
  return { name: req.data.name };
});

app.listen(() => {
  console.log("Eon running");
});
```

**Actual behavior**
Requesting `/` returns `{ name: "" }`.
Requesting `/:name` returns `{ name }`.

**Expected behavior**
Requesting `/` should result in 404.
Requesting `/:name` should return `{ name }` (as it does).

Node 12.14

---

I know you're trying to keep EON dependency-less, but why not use https://github.com/pillarjs/path-to-regexp.
It has 20+ millions of downloads/week, 100% test coverage with thousands of lines of test code.
https://bundlephobia.com/result?p=path-to-regexp@6.1.0 ~5kB

Or at least properly unit test your 50 line solution (in https://github.com/eon-web/eon/blob/master/engine/plugins/engine/pathresolve.js) properly.

**Comment by codemaster138** [comment]
path-to-regexp does look promising, but my main selling point is the whole 0dep-thing...

**Comment by codemaster138** [comment]
Maybe we should just try to build that kinda thing ourselves... it just looks like a parser that creates nodes that render to a regex. I have some experience with parsing, so I think that should be doable.

**Comment by codemaster138** [comment]
Opening a new branch `better_urls` for this

**Comment by codemaster138** [comment]
@dotvirus is there a spec for this path syntax?

**Comment by codemaster138** [comment]
> Or at least properly unit test your 50 line solution (in https://github.com/eon-web/eon/blob/master/engine/plugins/engine/pathresolve.js) properly.

@dotvirus How do you suggest we unit test our code without installing something like jest? The node.js assertion API is a nightmare

**Comment by codemaster138** [comment]
I guess devDependencies are OK though...

**Comment by dotvirus** [comment]
@codemaster138 Mocha, Chai, Ava, Jest. You can track code coverage with NYC (Istanbul).

As they say, don't reinvent the wheel. Why would I use a framework that has less test coverage on an essential part (like path matching).

Why not use path-to-regexp first to prototype and make it work properly, then some distant day when you feel like you're reaching its limits or it's too much of a black box, replace it with a custom solution. It's all about separation of concerns, and wasting time on trying to beat a well functioning and very well tested system, when really you're just trying to write a server framework, is not economical, at all.

**Comment by codemaster138** [comment]
@dotvirus your right... I guess we can get away with using only *essential* dependencies. I just don't want it to get out of hand. I feel like many people just want to go around installing a million things that don't really help that much just to make their life a bit easier when they could just spend 10 minutes and build something themselves, and I don't want eon to get like that. Like `vue-cli` with its 3000+ dependencies when all it needs to do for most people is create a Vue project

**Comment by codemaster138** [comment]
@dotvirus I'll create a version that uses this package on a different release tag so it doesn't interfere with the latest release until you've changed my mind about the necessity of using this package.

**Comment by codemaster138** [comment]
@dotvirus The version `v1.16.1-community` now supports `path-to-regexp`. It can be installed using
```bash
npm i eonjs@community
```
Community is a version that takes open suggestions from the community (hence the name) and allows itself to deviate slightly from these project rules like the whole 0-dependency thing.

The branch will be frequently rebased on master and a new version with all the changes will be pushed at every minor release of the main version.

Does this solve your problem? Feel free to inform me about your thoughts about this norm!

Also, a bugfix for the main version will be published shortly.

**Comment by codemaster138** [comment]
Pushed a new version `v1.16.2` that fixes #4.
